graphql.go

```diff
@@ -95,6 +113,13 @@ func (c *Client) do(ctx context.Context, op operationType, v any, variables map[
 			return err
 		}
 	}
+	if extensions != nil && out.Extensions != nil {
```

Comment:
Should we add unit tests that cover new lines?

Reply:
that could be another PR, but yeah